applications/luci-app-usteer/htdocs/luci-static/resources/view/usteer/usteer.js

```diff
@@ -91,25 +91,55 @@ function collectWlanAPInfoEntries(connectioninfo_table_entries, wlanAPInfos) {
 	}
 };
 
+function tootltip(mac, IP, hostname) {
+	var body= E([]);
+	var i=0;
+	body.appendChild(E('div', mac));
+	if (typeof IP !== 'undefined') {
+		while ((typeof IP['ipaddrs'][i] !== 'undefined') && (String(IP['ipaddrs'][i]).length > 0)) {
```

Comment:
Please revise these `while` loops to `forEach` loops. This makes them less ambiguous and avoids variable `i` declaration and assignment errors. You also avoid indices.

Reply:
I agree it is a bit messy, I made it into `for (var IPaddr in IP['ipaddrs']) body.appendChild(E('div', IP['ipaddrs'][IPaddr]));` now. So much cleaner. Thx